pkg/reconcilers/base_reconciler.go

```diff
@@ -0,0 +1,137 @@
+package reconcilers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var log = logf.Log.WithName("reconcilers")
```

Comment:
What's the advantage of creating a logger here as a package-level variable compared to receiving the one from the controller?
Does it also have some impact when dealing with testing? (possible mocking of the log etc...)

Reply:
Yes, this is a mistake. I will refactor a little bit.